### DISCUSSION TITLE
work on 6.11/6.12?

### DISCUSSION THREAD

**Discussion by n0xena** [openzfs/zfs#16574] [General]
// update - please see https://github.com/openzfs/zfs/discussions/16574#discussioncomment-11360884
// original
As I wasn't able to find issues or PRs for 6.11 this is just an informational question: What's the current status of work on Linux 6.11?

**Comment by robn** [comment]
Experimental support in #16400 has been merged; it's in 2.2.6 though not officially supported. It seems to mostly work though there's still some things we're chasing down (#16501 #16539 #16541).

**Reply by n0xena** [reply]
Thanks for the update.

**Comment by Thorazin** [comment]
I wonder if there are any updates with regards to compatibility with the 6.11 Kernel, and 6.12 for that matter? I see there is work being done, is there a timeline that it may be ready within?

**Reply by n0xena** [reply]
vanilla arch here
I run 2.2.6 on 6.11.x pretty much from the start and recently updated to 2.3.0-rc2 and then 2.3.0-rc3
I just tried to build 2.2.6 and 2.3.0-rc3 - but 2.2.6 fails durin build and 2.3.0 already stops at configure with the enforcement of kernel version

**Comment by IvanVolosyuk** [comment]
There is zfs-2.3.0-rc3 with linux 6.11 support.

**Comment by n0xena** [comment]
So as I run 2.2.6 and then 2.3.0-rc2/-rc3 on 6.11.x quite stable for some time I just tried to build for 6.12 - and it fails
2.2.6 fails somewhere during the build - 2.3.0-rc3 already stops during configure with the new kernel version enforcement
so - to update the question: with 6.11 working pretty much since 2.2.6 (which had unofficial support) - how about 6.12? It was just released on arch and as last release of the year it'll become the new lts - so "stick to lts" doesn't cut it anymore when it actually means "stick to 6.6" as "lts" now becomes 6.12.
Is 6.12 planed for 2.3.0? Will it be backported to 2.2.x and 2.1.x? And what's the further plan for 2.1 and 2.2 anyway given there're already up-to-date patchset PRs?

**Reply by IvanVolosyuk** [reply]
**RE: 6.12 fails to build:** 
You can build 2.3.0-rc3 against 6.12 with ```--enable-linux-experimental``` configure flag.

```
        You are building OpenZFS against Linux version 6.12.0-gentoo-x86_64.

        This combination is considered EXPERIMENTAL by the OpenZFS project.
        Even if it appears to build and run correctly, there may be bugs that
        can cause SERIOUS DATA LOSS.

        YOU HAVE BEEN WARNED!

        If you choose to continue, we'd appreciate if you could report your
        results on the OpenZFS issue tracker at:

            https://github.com/openzfs/zfs/issues/new

        Your feedback will help us prepare a new OpenZFS release that supports
        this version of Linux.
```
I built the kernel with '--enable-linux-builtin --enable-linux-experimental' successfully, though I hit #16620 when booting in VM.

**Reply by n0xena** [reply]
Thanks for that information. I was able to build [zfs-linux-rc-2.3.0_rc3_6.12.1.arch1.1-1-x86_64.pkg.tar.zst](https://github.com/n0xena/archzfs/releases/tag/2.3.0-rc3_6.12.1) and will test it now.

**Comment by Thorazin** [comment]
ok so, how do I build 2.3 for kernel 6.11? I ask because following the directions [here](https://openzfs.github.io/openzfs-docs/Developer%20Resources/Custom%20Packages.html#get-the-source-code) for a rpm dkms build led to errors on Fedora 41.

**Reply by n0xena** [reply]
the linked instruction seems straight forward:
make sure the build depencies are installed
download rc3 tar
extract
run ./configure
run make rpm-utils rpm-dkms
if you encounter errors please open a issue yourself - I represent and refer to the archlinux project - can't help with fedora
